tensorflow/python/distribute/input_lib.py

```diff
@@ -1590,18 +1606,34 @@ def __init__(self, dataset=None, worker=None, devices=None, components=None,
     else:
       if (components is not None or element_spec is not None):
         raise ValueError(error_message)
-      super(_SingleWorkerOwnedDatasetIterator, self).__init__(dataset, worker,
-                                                              devices)
+      super(_SingleWorkerOwnedDatasetIterator, self).__init__(dataset=dataset,
+                                                              worker=worker,
+                                                              devices=devices)
 
   def _make_iterator(self):
     """Make appropriate iterator on the dataset."""
     if not self._worker:
       raise ValueError("Worked device must be specified when creating an "
                        "owned iterator.")
-    host_device = device_util.get_host_for_device(self._worker)
-    with ops.device(self._worker):
-      self._iterator = multi_device_iterator_ops.OwnedMultiDeviceIterator(
-          self._dataset, self._devices, source_device=host_device)
+    if (self._options is None
+            or self._options.experimental_replication_mode == InputReplicationMode.PER_WORKER):
+      host_device = device_util.get_host_for_device(self._worker)
+      with ops.device(self._worker):
+        self._iterator = multi_device_iterator_ops.OwnedMultiDeviceIterator(
+            self._dataset, self._devices, source_device=host_device)
+    else:
+      if self._options.experimental_copy_dataset_on_device:
+        worker_device = self._devices[0]
+      else:
+        worker_device = self._worker
+      if self._options.experimental_prefetch_to_device:
+        host_device = device_util.get_host_for_device(self._worker)
```

Comment:
Now looking through this, on second thoughts, I am not sure if this case makes sense:
experimental_copy_dataset_on_device=True and experimental_prefetch_to_device=True

In this case, what is "host device"? i m not clear.. i think OwnedMultiDeviceIterator may not even work when placed on the GPU. 

Can we actually just throw an error in this case? (Can do it earlier on, near line 140)

Reply:
I tested and the op is not implemented. Added the error in both places